### PR TITLE
Fixed video hashing and applied consistent formatting

### PR DESCRIPTION
**Pull request by timstableford**
* GetNumberVideoFrames and fps functions in cimgffmpeg.cpp didn't correctly allocate and free their AVFormatContext structures causing segfaults when hashing videos.
* The usage of TestVideoHash was not particularly intuitive, nor did the code flow well. Hopefully my changes here didn't break anybodies workflows.
* Applied clang-format to the code-base, the inconsistent styling made it difficult to understand.
* Added a .gitignore file.

### CODE REVIEW

**Comment by timstableford** [comment]
There's a build failure but I don't think it's me?

/Users/travis/build/aetilius/pHash/third-party/CImg/CImg.h:406:10: fatal error: 'X11/Xlib.h' file not found

#include <X11/Xlib.h>

**Comment by timstableford** [comment]
This may still need work. Getting suspiciously short hashes back

**Comment by timstableford** [comment]
I did accidentally change the behaviour of getting the number of frames but the short hashes seem to be expected.

**Comment by eklinger** [comment]
> There's a build failure but I don't think it's me?
> 
> /Users/travis/build/aetilius/pHash/third-party/CImg/CImg.h:406:10: fatal error: 'X11/Xlib.h' file not found
> 
> #include <X11/Xlib.h>

I believe you need to ensure #define cimg_display 0. It should be done by cmake in the generated pHash.h header

**Comment by simplyboo6** [comment]
Thanks @eklinger I think that's fixed but there seems to be errors with brew upgrade now.

File: examples/test_dctvideohash.cpp [hunk 1]
```diff
@@ -22,68 +22,51 @@
...
     }
-    const char *file1 = argv[1];
-    const char *file2t = argv[2];
 
-    char *file2 = strdup(file2t);
-
-    if (!file1 || !file2){
-	printf("not enough arguments\n");
-	exit(1);
+    ulong64 **hashes = (ulong **)malloc(sizeof(ulong64 *) * (argc - 1));
```
Comment (aetilius) [inline examples/test_dctvideohash.cpp]: Please correct to ulong64

Comment (timstableford) [reply examples/test_dctvideohash.cpp]: Done

**Pull request merged**